app/controllers/sign_up/completions_controller.rb

```diff
@@ -89,5 +90,14 @@ def pii
       pii_string = Pii::Cacher.new(current_user, user_session).fetch_string
       JSON.parse(pii_string || '{}', symbolize_names: true)
     end
+
+    def send_in_person_completion_survey
+      return unless sp_session_ial == 2
```

Comment:
what is the significance of 2 here?

Reply:
This requires that the user be sharing attributes verified via [IAL2](https://pages.nist.gov/800-63-3/sp800-63a.html#44-identity-assurance-level-2) with the partner in order to trigger the survey email (as opposed to [IAL1](https://pages.nist.gov/800-63-3/sp800-63a.html#43-identity-assurance-level-1) or [IAL3](https://pages.nist.gov/800-63-3/sp800-63a.html#-45-identity-assurance-level-3)). Note that this current usage excludes the IAL "strict" mode.

This value is passed as [an attribute from the partner](https://developers.login.gov/attributes/). The value then [gets mapped](https://github.com/18F/identity-idp/blob/b22458b587095d47011416ce4e92f7b1a00ec360/lib/saml_idp_constants.rb#L37) to [the IAL constants](https://github.com/18F/identity-idp/blob/b22458b587095d47011416ce4e92f7b1a00ec360/lib/idp/constants.rb#L11) (in [a session value](https://github.com/18F/identity-idp/blob/b22458b587095d47011416ce4e92f7b1a00ec360/app/services/store_sp_metadata_in_session.rb#L38)) during [authentication](https://github.com/18F/identity-idp/blob/b22458b587095d47011416ce4e92f7b1a00ec360/app/services/service_provider_request_handler.rb#L15).